Apps/WebClient/src/ClientApp/src/components/dependentCard.vue

```diff
@@ -0,0 +1,254 @@
+    addError!: (error: BannerError) => void;
+
+    private isLoading = false;
+    private logger!: ILogger;
+    private laboratoryService!: ILaboratoryService;
+    private labResults: LaboratoryOrder[] = [];
+    private isDataLoaded = false;
+
+    private get isExpired() {
+        let birthDate = new DateWrapper(this.dependent.dateOfBirth);
+        let now = new DateWrapper();
+        return now.diff(birthDate, "year").years > 19;
+    }
+
+    private mounted() {
+        this.logger = container.get<ILogger>(SERVICE_IDENTIFIER.Logger);
+        this.laboratoryService = container.get<ILaboratoryService>(
+            SERVICE_IDENTIFIER.LaboratoryService
+        );
+    }
+
+    private fetchLaboratoryResults() {
+        if (this.isDataLoaded) {
+            return;
+        }
+        this.isLoading = true;
+        this.getLaboratoryOrders({ hdid: this.dependent.hdid })
+            .then((results) => {
+                if (results.resultStatus == ResultType.Success) {
+                    this.labResults = results.resourcePayload;
+                    this.isDataLoaded = true;
+                } else {
+                    this.logger.error(
+                        "Error returned from the laboratory call: " +
+                            JSON.stringify(results.resultError)
+                    );
+                    this.addError(
+                        ErrorTranslator.toBannerError(
+                            "Fetch Laboratory Error",
+                            results.resultError
+                        )
+                    );
+                }
+            })
+            .catch((err) => {
+                this.logger.error(err);
+                this.addError(
+                    ErrorTranslator.toBannerError("Fetch Laboratory Error", err)
+                );
+            })
+            .finally(() => {
+                this.isLoading = false;
+            });
+    }
+
+    private getReport(labOrder: LaboratoryOrder) {
+        let labResult = labOrder.labResults[0];
+        this.laboratoryService
+            .getReportDocument(labResult.id, this.user.hdid)
+            .then((result) => {
+                const link = document.createElement("a");
+                let report: LaboratoryReport = result.resourcePayload;
+                link.href = `data:${report.mediaType};${report.encoding},${report.data}`;
+                link.download = `COVID_Result_${this.dependent.name}_${labResult.collectedDateTime}.pdf`;
+                link.click();
+                URL.revokeObjectURL(link.href);
+            })
+            .catch((err) => {
+                this.logger.error(err);
+                this.addError(
+                    ErrorTranslator.toBannerError(
+                        "Download Laboratory Report Error",
+                        err
+                    )
+                );
+            });
+    }
+
+    private formatDate(date: StringISODate): string {
```

Comment:
I think we have this refactored somewhere on vue side, we can just reuse the existing one.

Reply:
i'll look into that

Reply:
@leobcgov can you point me to where? i couldn't find it and i was unable to create a solution that didn't require a local format method in the vue file. even if I move the method to a separate file it always requires a local function to call the external method.

Reply:
@tiagograf-AOT I may be wrong as we may have a different formatDate function. We can leave it as is then refactor it in the future then